libs/anglify/src/modules/input/input.directive.ts

```diff
@@ -1,69 +1,111 @@
-import { Directive, ElementRef, OnInit, Optional, Self } from '@angular/core';
+import { Directive, ElementRef, Input, OnInit, Optional, Self } from '@angular/core';
 import { NgControl } from '@angular/forms';
 import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
-import { fromEvent, merge, NEVER, Observable, of, Subject } from 'rxjs';
-import { debounceTime, distinctUntilChanged, map, shareReplay, startWith, switchMap } from 'rxjs/operators';
-import { observeOnMutation } from '../../../utils/functions';
+import {
+  debounceTime,
+  distinctUntilChanged,
+  fromEvent,
+  map,
+  merge,
+  NEVER,
+  Observable,
+  of,
+  share,
+  startWith,
+  Subject,
+  switchMap,
+} from 'rxjs';
+import { AnglifyIdService } from '../../services/id/id.service';
+import { observeOnMutation } from '../../utils/functions';
 
 @UntilDestroy()
 @Directive({
   selector: 'input[anglifyInput], textarea[anglifyInput]',
 })
 export class InputDirective implements OnInit {
-  private readonly nativeElement = this.elementRef.nativeElement;
-  private readonly mutationObserver$ = observeOnMutation(this.nativeElement, { attributes: true }).pipe(shareReplay(1));
-  private readonly inputEvent$ = merge(fromEvent(this.nativeElement, 'input'), this.mutationObserver$).pipe(shareReplay(1));
-  private readonly statusChanged$ = new Subject();
+  public readonly id = this.idService.generate();
 
-  public readonly length$ = this.inputEvent$.pipe(
+  @Input() public set rows(rows: number) {
+    // @ts-expect-error
+    this.elementRef.nativeElement.rows = rows;
```

Comment:
Is this because of the union type in `ElementRef<HTMLTextAreaElement | HTMLInputElement>`?

If so you should probably type guard it instead of ignoring, or making sure its a TextArea Element:

```suggestion
    const element = this.elementRef.nativeElement;
    if (element instanceof HTMLTextAreaElement) {
      element.rows = rows
    }
```